backend/src/services/certificate-v3/certificate-v3-service.ts

```diff
@@ -479,9 +637,342 @@ export const certificateV3ServiceFactory = ({
     });
   };
 
+  const renewCertificate = async ({
+    certificateId,
+    actor,
+    actorId,
+    actorAuthMethod,
+    actorOrgId,
+    internal = false
+  }: TRenewCertificateDTO & { internal?: boolean }): Promise<TCertificateFromProfileResponse> => {
+    const originalCert = await certificateDAL.findById(certificateId);
+    if (!originalCert) {
+      throw new NotFoundError({ message: "Certificate not found" });
+    }
+
+    if (!originalCert.profileId) {
+      throw new ForbiddenRequestError({
+        message: "Only certificates issued from a profile can be renewed"
+      });
+    }
+
+    const originalSignatureAlgorithm = originalCert.signatureAlgorithm as CertSignatureAlgorithm;
+    const originalKeyAlgorithm = originalCert.keyAlgorithm as CertKeyAlgorithm;
+
+    if (!originalSignatureAlgorithm || !originalKeyAlgorithm) {
+      throw new BadRequestError({
+        message:
+          "Original certificate does not have algorithm information stored. Cannot renew certificate issued before algorithm tracking was implemented."
+      });
+    }
+
+    const profile = await certificateProfileDAL.findByIdWithConfigs(originalCert.profileId);
+    if (!profile) {
+      throw new NotFoundError({ message: "Certificate profile not found" });
+    }
+
+    if (profile.enrollmentType !== "api") {
+      throw new ForbiddenRequestError({
+        message: "Certificate is not eligible for renewal: EST certificates cannot be renewed through this endpoint"
+      });
+    }
+
+    const ca = await certificateAuthorityDAL.findByIdWithAssociatedCa(profile.caId);
+    if (!ca) {
+      throw new NotFoundError({ message: "Certificate Authority not found" });
+    }
+
+    const eligibilityCheck = validateRenewalEligibility(originalCert, ca);
+    if (!eligibilityCheck.isEligible) {
+      throw new BadRequestError({
+        message: `Certificate is not eligible for renewal: ${eligibilityCheck.errors.join(", ")}`
+      });
+    }
+
+    if (!internal) {
+      const { permission } = await permissionService.getProjectPermission({
+        actor,
+        actorId,
+        projectId: profile.projectId,
+        actorAuthMethod,
+        actorOrgId,
+        actionProjectType: ActionProjectType.CertificateManager
+      });
+
+      ForbiddenError.from(permission).throwUnlessCan(
+        ProjectPermissionCertificateProfileActions.IssueCert,
+        ProjectPermissionSub.CertificateProfiles
+      );
+    }
```

Comment:
it would be best if we did this right after we fetched the profile so that there's not much details we're exposing in terms of errors